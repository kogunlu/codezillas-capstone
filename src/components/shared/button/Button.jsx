/* eslint-disable react/destructuring-assignment */
import { useNavigate } from "react-router-dom";

export default function Button(props)
{
    const navigate = useNavigate();

    function handleClick()
    {
        if(props.function)
        {
            props.function()
        }else{
            navigate('/')
        }
    }
    return(
        <input 
        type={props.type ? props.type : "button"}
        value={props.name ? props.name : "CLICK"}
        className={props.classes ? props.classes : " py-2 px-5  hover:bg-cyan-200 bg-cyan-400 md:font-medium md:font-small rounded text-sm md:text-lg"}
        onClick={()=> handleClick()} />
    );
}