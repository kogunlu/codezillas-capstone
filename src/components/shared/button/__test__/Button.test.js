import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Button from '../Button';

test('Navbar component renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><Button/></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
