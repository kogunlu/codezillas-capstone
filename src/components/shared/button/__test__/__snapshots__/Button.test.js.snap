// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<input
  className=" py-2 px-5  hover:bg-cyan-200 bg-cyan-400 md:font-medium md:font-small rounded text-sm md:text-lg"
  onClick={[Function]}
  type="button"
  value="CLICK"
/>
`;
